### PR TITLE
SUBMARINE-978. Add model management database service.

### PR DESCRIPTION
**Pull request by jeff-901**
### What is this PR for?
Create model service for later serving and other model management REST api.

### What type of PR is it?
Feature

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-978

### How should this be tested?
Call the model service api and see the difference on the mlflow UI.
If you test locally edit the jdbc url ip address in ModelBatisUtil.java  and modelbatis-config.xml.
I will add formal test on https://issues.apache.org/jira/browse/SUBMARINE-983.

### Questions:
* Do the license files need updating? No
* Are there breaking changes for older versions? No
* Does this need new documentation? No

### CODE REVIEW

**Comment by jeff-901** [comment]
@ByronHsu @KUAN-HSUN-LI help me review the code. Thanks.

File: submarine-server/server-core/src/main/resources/org/apache/submarine/database/mappers/ModelVersionMapper.xml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    <result column="version" property="version" />
+    <result column="creation_time" property="createTime" />
+    <result column="last_updated_time" property="lastUpdatedTime" />
+    <result column="description" property="description" />
+    <result column="user_id" property="userId" />
+    <result column="current_stage" property="currentStage" />
+    <result column="source" property="source" />
+    <result column="run_id" property="runId" />
+    <result column="status" property="status" />
+    <result column="status_message" property="statusMessage" />
```
Comment (KUAN-HSUN-LI) [inline submarine-server/server-core/src/main/resources/org/apache/submarine/database/mappers/ModelVersionMapper.xml:33]: @jeff-901 I am confused with `current_stage` and `status`. What is the difference between these two data? Thanks

Comment (jeff-901) [reply submarine-server/server-core/src/main/resources/org/apache/submarine/database/mappers/ModelVersionMapper.xml:33]: @KUAN-HSUN-LI  `current_stage` indicates the model is Staging, Production or Archived. `status` shows whether the model is ready.

Comment (jeff-901) [reply submarine-server/server-core/src/main/resources/org/apache/submarine/database/mappers/ModelVersionMapper.xml:33]: @KUAN-HSUN-LI  `current_stage` indicates the model is Staging, Production or Archived. `status` shows whether the model is ready.

Comment (KUAN-HSUN-LI) [reply submarine-server/server-core/src/main/resources/org/apache/submarine/database/mappers/ModelVersionMapper.xml:33]: @jeff-901  Thanks for your explanation.

**Review by KUAN-HSUN-LI — APPROVED** [review]
LGTM

**Review by pingsutw — APPROVED** [review]
+1

**Pull request closed without merging**